config/management/operational/src/validator_config.rs

```diff
@@ -55,32 +55,45 @@ impl SetValidatorConfig {
         let client = JsonRpcClientWrapper::new(config.json_server);
         let sequence_number = client.sequence_number(operator_account)?;
 
-        // Retrieve the current validator / fullnode addresses and update accordingly
-        let vc = client.validator_config(owner_account)?;
-        let validator_config = DecryptedValidatorConfig::from_validator_config_resource(
-            &vc,
-            owner_account,
-            &encryptor,
-        )
-        .map_err(|e| Error::UnexpectedError(format!("Error parsing validator config: {}", e)))?;
+        // See if the validator is in the set
+        let in_set = client
+            .validator_set(None)?
+            .iter()
+            .any(|vi| vi.account_address() == &owner_account);
+
+        let validator_config = if in_set {
+            // Retrieve the current validator / fullnode addresses and update accordingly
+            let vc = client.validator_config(owner_account)?;
+            DecryptedValidatorConfig::from_validator_config_resource(&vc, owner_account, &encryptor)
+                .map(Some)
+                .map_err(|e| {
+                    Error::UnexpectedError(format!("Error parsing validator config: {}", e))
+                })?
+        } else {
+            None
+        };
 
         let validator_address = if let Some(validator_address) = self.validator_address {
             validator_address
+        } else if let Some(vc) = &validator_config {
+            strip_address(&vc.validator_network_address)
         } else {
-            strip_address(&validator_config.validator_network_address)
+            "/ip4/127.0.0.1/tcp/6180".parse().unwrap()
```

Comment:
Can you document why we want these to be the defaults? Or is this related to the debugging you did locally?

Reply:
Did you and @gregnazario work together on reviewing this, rofl... I'll raise an error.